[DOCS] Blocklist (#1811)
elastic/security-docs: docs/management/admin/blocklist.asciidoc

[[blocklist]]
[chapter]
= Blocklist

coming[8.2.0]

> The blocklist allows you to prevent specified applications from running on hosts, extending the list of processes that {endpoint-sec} considers malicious. This is especially useful to ensure that known malicious processes aren't accidentally executed by an end user. 

Author: ```suggestion
The blocklist allows you to prevent specified applications from running on hosts, extending the list of processes that {endpoint-sec} considers malicious. This is especially useful for ensuring that known malicious processes aren't accidentally executed by end users. 
```

> The blocklist allows you to prevent specified applications from running if you're certain they should never run in a host environment, extending the list of processes that {endpoint-sec} considers malicious. This is especially useful to ensure that known malicious processes aren't accidentally executed by an end user. 

Reviewer: Not sure if the last sentence (the one I made suggestions on) is necessary, I think folks are pretty clear on the value of blocklisting and it extends beyond adding guardrails for users/insiders.

> The blocklist allows you to prevent specified applications from running if you're certain they should never run in a host environment, extending the list of processes that {endpoint-sec} considers malicious. This is especially useful to ensure that known malicious processes aren't accidentally executed by an end user. 

Author: I get what you mean, but I also like having a little extra explanation for the feature beyond the boilerplate description. @caitlinbetz, any thoughts? Or other use case info you think would be helpful here?

This combines @benironside's edits and other tweaks for consistency with similar wording:
```suggestion
The blocklist allows you to prevent specified applications from running on hosts, extending the list of processes that {endpoint-sec} considers malicious. This is especially useful for ensuring that known malicious processes aren't accidentally executed by end users.
```

> The blocklist allows you to prevent specified applications from running if you're certain they should never run in a host environment, extending the list of processes that {endpoint-sec} considers malicious. This is especially useful to ensure that known malicious processes aren't accidentally executed by an end user. 

Author: ```suggestion
The blocklist allows you to prevent specified applications from running on hosts, extending the list of processes that {endpoint-sec} considers malicious. This is especially useful to ensure that known malicious processes aren't accidentally executed by an end user. 
```

[NOTE] 
=====
In addition to configuring specific entries on the **Blocklist** page, you must also ensure that the blocklist is enabled on the {endpoint-sec} integration policy in the <<malware-protection, Malware protection settings>>. This setting is enabled by default.

> You must have the built-in `superuser` role to access the blocklist. For more information, refer to {ref}/built-in-users.html[Built-in users].

Author: Since most of the pages for endpoint management are so similar, I'd like to make this update in a separate issue that includes adding the prerequisites box consistently across all the pages. I was already thinking of creating an issue for general consistency fixes, so I think the prereqs box would fall nicely into that issue's scope. I'll add a link to this PR once I've created the other issue.

> You must have the built-in `superuser` role to access this feature. For more information, refer to {ref}/built-in-users.html[Built-in users].

Author: I think it's actually both, but it's intended to refer to just the blocklist feature.
```suggestion
You must have the built-in `superuser` role to access the blocklist. For more information, refer to {ref}/built-in-users.html[Built-in users].
```

=====

> By default, a blocklist entry is recognized globally across all hosts running {endpoint-sec}. If you have a https://www.elastic.co/pricing[Platinum or Enterprise subscription], you can also assign a blocklist entry to a specific {endpoint-sec} integration policy, which blocks the process only on hosts assigned to that policy.

Author: I think this should remain mostly singular, because it's talking about configuring things on a per-entry basis (not across multiple entries). But a single blocklist entry (representing a single process) can be assigned to multiple policies, so we can make policies plural:
```suggestion
By default, a blocklist entry is recognized globally across all hosts running {endpoint-sec}. If you have a https://www.elastic.co/pricing[Platinum or Enterprise subscription], you can also assign a blocklist entry to specific {endpoint-sec} integration policies, which blocks the process only on hosts assigned to that policy.
```

. Go to **Manage** -> **Blocklist**.

. Click **Add blocklist entry**. The **Add blocklist** flyout is displayed.

. Fill in these fields in the **Details** section:
.. `Name`: Enter a name to identify the blocklist entry.
.. `Description`: Enter a description to provide more information on the blocklist entry (optional).

. In the **Conditions** section, enter the following information about the application you want to block:
.. `Select operating system`: Select the appropriate operating system from the drop-down.
.. `Field`: Select a field to identify the application being blocked:
  * `Hash`: The MD5, SHA-1, or SHA-256 hash value of the application's executable.
  * `Path`: The full file path of the application's executable.
  * `Signature`: (Windows only) The name of the application's digital signer.
+
TIP: To find the signer's name for an application, go to *Kibana* -> *Discover* and query the process name of the application's executable (for example, `process.name : "mctray.exe"` for a McAfee security binary). Then, search the results for the `process.code_signature.subject_name` field, which contains the signer's name (for example, `McAfee, Inc.`).

> .. `Operator`: The operator is `is one of`, and it cannot be modified.

Reviewer 2: ```suggestion
.. `Operator`: The operator is `is one of` and cannot be modified.
```

.. `Value`: Enter the hash value, file path, or signer name. To enter multiple values (such as a list of known malicious hash values), you can enter each value individually or paste a comma-delimited list, then press **Return**.
+

> NOTE: Hash values must be valid to add them to the blocklist. In addition, to minimize visibility gaps in the {security-app}, be as specific as possible in your entries. For example, combine `Signature` information with a known `Path`.

Author: Excellent catch -- I copied a lot of these steps from the trusted applications feature because they're so similar, and I should have cut this out. Trusted apps basically do the opposite of blocklist, hence why this seems (rightly) backwards to you!
```suggestion
NOTE: Hash values must be valid to add them to the blocklist.
```

. Select an option in the *Assignment* section to assign the blocklist entry to a specific integration policy:
+
* `Global`: Assign the blocklist entry to all integration policies for {endpoint-sec}.
* `Per Policy`: Assign the blocklist entry to one or more specific {endpoint-sec} integration policies. Select each policy where you want the blocklist entry to apply.
+
NOTE: You can also select the `Per Policy` option without immediately assigning a policy to the blocklist entry. For example, you could do this to create and review your blocklist configurations before putting them into action with a policy.

. Click **Add blocklist**. The new entry is added to the **Blocklist** page.

> . When you're done adding entries to the blocklist, make sure that the blocklist is enabled for the {endpoint-sec} integration policy:

Reviewer 3: My vote is updating `make sure` ==> `ensure`.

> . When you're done adding entries to the blocklist, make sure that the blocklist is enabled for the {endpoint-sec} integration policy:

Author: Looking at `an` vs. `the` here, I realized this needs a bit more explanation:
```suggestion
. When you're done adding entries to the blocklist, ensure that the blocklist is enabled for the {endpoint-sec} integration policies that you just assigned:
```

.. Go to **Manage** -> **Policies**, then click on an integration policy.

> .. On the **Policy settings** tab, make sure that the **Malware protections enabled** and **Blocklist enabled** toggles are switched on. Both settings are enabled by default.

Reviewer 3: ```suggestion
.. On the **Policy settings** tab, ensure that the **Malware protections enabled** and **Blocklist enabled** toggles are switched on. Both settings are enabled by default.
```

[discrete]
[[manage-blocklist]]
== View and manage the blocklist

The *Blocklist* page displays all the blocklist entries that have been added to the {security-app}. To refine the list, use the search bar to search by name, description, or a field value.

[role="screenshot"]
image::images/blocklist.png[]

[discrete]
[[edit-blocklist-entry]]
=== Edit a blocklist entry

> You can individually modify each blocklist entry. With a Platinum or Enterprise subscription, you can also change the policies applied to a blocklist entry.

Reviewer: Not sure I captured your intent here but I wasn't sure what it would mean to apply policies to a blocklist entry.

> You can individually modify each blocklist entry. With a Platinum or Enterprise subscription, you can also change the policies applied to a blocklist entry.

Author: Right, it's the other way around (per the UI copy, a blocklist entry is applied/assigned to one or more policies). Maybe this is a good spot for a dangling preposition:
```suggestion
You can individually modify each blocklist entry. With a Platinum or Enterprise subscription, you can also change the policies that a blocklist entry is assigned to.
```

> You can individually modify each blocklist entry. With a Platinum or Enterprise subscription, you can also change the policies applied to a blocklist entry.

Reviewer: 💯 💯  💯

To edit a blocklist entry:

. Click the actions button (*...*​) for the blocklist entry you want to edit, then select *Edit blocklist*.
. Modify details as needed.
. Click *Save*.

[discrete]
[[delete-blocklist-entry]]
=== Delete a blocklist entry

> You can delete a blocklist entry, which removes it entirely from all {endpoint-sec} policies.

Reviewer 3: Suggestion - feel free to reword. 

```suggestion
You can delete a blocklist entry, which removes it entirely from all {endpoint-sec} policies and allows end users to access the file or application that was previously blocked.
```

Author: ```suggestion
You can delete a blocklist entry, which removes it entirely from all {endpoint-sec} policies. This allows end users to access the application that was previously blocked.
```

To delete a blocklist entry:

. Click the actions button (*...*) for the blocklist entry you want to delete, then select *Delete blocklist*.

> . On the dialog that opens, verify that you are removing the correct blocklist entry, then click *Delete*. A confirmation message is displayed.

Reviewer 3: Aiming for active voice, where possible.
```suggestion
. On the dialog that opens, verify that you are removing the correct blocklist entry, then click *Delete*. A confirmation message displays.
```
